core/src/utils/wiremock_utils.rs

```diff
@@ -126,6 +135,46 @@ pub async fn setup_wiremock() -> String {
         .mount(&mock_server)
         .await;
 
+    // block_transaction_count hash
+    let block_id = BlockId::Hash(
+        H256::from_str("0x0449aa33ad836b65b10fa60082de99e24ac876ee2fd93e723a99190a530af0a9")
+            .unwrap(),
+    );
+    let starknet_block_id = ethers_block_id_to_starknet_block_id(block_id).unwrap();
+    Mock::given(method("POST"))
+        .and(body_json(StarknetRpcBaseData::get_block_transaction_count(
+            [&starknet_block_id],
+        )))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_raw(
+                    include_str!("data/blocks/starknet_getBlockTransactionCount.json"),
+                    "application/json",
+                )
+                .append_header("vary", "Accept-Encoding")
+                .append_header("vary", "Origin"),
+        )
+        .mount(&mock_server)
+        .await;
+
+    // block_transaction_count latest
+    let latest_block = StarknetBlockId::Tag(BlockTag::Latest);
+    Mock::given(method("POST"))
+        .and(body_json(StarknetRpcBaseData::get_block_transaction_count(
+            [&latest_block],
+        )))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_raw(
+                    include_str!("data/blocks/starknet_getBlockTransactionCount.json"),
```

Comment:
did you add this file?

since it's not a `starknet_getBlockTransactionCount` call we're doing inside our core logic, we don't have to call it that, we don't even have to mock it,

what we're really mocking is `client.get_block_with_txs` from [this line](https://github.com/sayajin-labs/kakarot-rpc/pull/108/files#diff-e6d20c367d720630c43945964dc7fdb3105ee5c321f359ce26f51c44b820f68aR449)

which means we have to "format" well a call to get_block_with_txs if it's not already being mocked properly,
if tests pass etc. maybe an already existing get_block_with_txs mock is enough for our use case of filtering txs

Reply:
You were right, the mock was eliminated because it was not being used.